src/main/java/org/apache/maven/plugins/install/InstallMojo.java

```diff
@@ -153,31 +163,90 @@ private boolean allProjectsMarked()
         return true;
     }
 
-    private void installProject( MavenProject pir )
-        throws MojoFailureException, MojoExecutionException
+    private void installProject( MavenProject project ) throws MojoExecutionException, MojoFailureException
     {
         try
         {
-            installer.install( session.getProjectBuildingRequest(), new ProjectInstallerRequest().setProject( pir ) );
+            repositorySystem.install( session.getRepositorySession(), processProject( project ) );
         }
-        catch ( IOException e )
+        catch ( IllegalArgumentException e )
         {
-            throw new MojoFailureException( "IOException", e );
+            throw new MojoFailureException( e.getMessage(), e );
```

Comment:
why not `MojoExecutionException`

I must find a discussion, but as I remember `MojoFailureException` should not be used ....

Reply:
Hm, we need to clean up this I guess.... (as now I see that I used the two inconsistently).

Reply:
I've started discussion on dev list about it 😄